### PR TITLE
Raise test coverage from 86% to 97%

### PR DESCRIPTION
**Pull request by maxime-leiber**
## Summary

Nightly-agent task #6 (see `pytest --cov=src/dstft --cov-report=term-missing`, was 86%).

- Adds `tests/test_core.py`: direct unit tests for `dstft._core`'s internal validation branches (`overlap_add_dual`, `overlap_add_dual_dft_wola_den`, `fdstft_fft_forward`, `adstft_dft_forward`, `adstft_dft_adjoint`, `cg_solve`). These functions are only ever called by `DSTFT.forward()`/`inverse()` with already-consistent shapes, so their defensive `raise` branches were unreachable through the public API — that's what the coverage gap was actually flagging. `_core.py` is now at 100%.
- Removes `overlap_add_wola` from `_core.py`: a WOLA synthesis function superseded by `overlap_add_dual`, never called anywhere (grepped across `src/` and `__init__.py`'s `__all__`) and not part of the public API — dead code, not a behavior change.
- Marks `DSTFT._inverse_dft_exact` with `# pragma: no cover` + a comment: it's a stub for a not-yet-wired-in future exact DFT-backend inverse (raises `NotImplementedError` unconditionally). Kept rather than deleted since its docstring signals ongoing/future work.
- Adds `initialize()`'s own non-tensor/wrong-ndim input validation tests (previously only `forward()`'s equivalent checks were tested).
- Adds `plot_spec`/`plot_win_lengths` coverage for `show=True` and axis-reuse + `title` branches.

**Result: 86% → 97%** (`_core.py`, `visualization.py`, `windows.py`: 100%; `dstft.py`: 94%).

The remaining `dstft.py` gaps are internal invariant guards (e.g. "initialization metadata is missing" checked right after `_initialized` is confirmed `True`) that the class's own logic should make unreachable through the public API without directly corrupting private state — plus one device-mismatch `RuntimeError` (re-`initialize()` on a different device) that's untestable here without a second real device. Writing artificial tests that reach into private attributes to hit these felt like the wrong trade — documenting them here instead, per the task's own guidance.

**Minor finding along the way (not fixed, left for review):** in `overlap_add_dual`, a window with a broadcastable leading dim (`shape[0] == 1`) but a mismatched frame-count second dim raises `RuntimeError` from the underlying `.expand()` call instead of the documented `ValueError` — see the comment on `test_overlap_add_dual_rejects_window_frame_count_mismatch`. Not exercised by any current caller in `dstft.py` (it always passes consistent shapes), so this is a latent robustness gap rather than an active bug. Flagging since it touches `_core.py`'s numeric-path code and is Maxime's call whether it's worth a follow-up.

## Test plan

- [x] `pytest --cov=src/dstft --cov-report=term-missing` — 115 passed, 1 skipped (CUDA), 97% overall coverage
- [x] `pre-commit run --all-files` — clean (ruff, ruff-format, mypy, interrogate)

---
_Generated by [Claude Code](https://claude.ai/code/session_01KyjjDHc4vpTx6jDv1rWfC7)_

## Summary by CodeRabbit

* **Breaking Changes**
  * Removed the public `overlap_add_wola` synthesis function.

* **Bug Fixes**
  * Improved validation for transform, overlap-add, windowing, and solver operations.
  * Added clearer handling of invalid initialization inputs.

* **Tests**
  * Expanded coverage for validation, convergence, reconstruction output, and visualization behavior.
  * Added checks for plotting figures, axes, titles, axis reuse, and non-interactive display.

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+    # mismatched *second* dim instead raises a RuntimeError from the
+    # underlying `.expand()` call, not this ValueError - see PR description.
+    with pytest.raises(ValueError, match="frames dimension must be 1 or match frames"):
+        _core.overlap_add_dual(
+            frames=_frames(frames=3),
+            frame_positions=_positions(frames=3),
+            analysis_window=torch.ones(2, 3, 8),  # leading dim 2 != 1 and != 3
+            signal_length=16,
+            eps=1e-8,
+        )
```
[the remark this answers is not in the document]
Comment (maxime-leiber) [reply tests/test_core.py:122]: Good catch, and correct diagnosis. Deliberately not fixing this in `overlap_add_dual` (`src/dstft/_core.py`) here though: this is a numeric-path validation change to `src/dstft/`, and this repo's convention is that those stay for Maxime's own review regardless of CI/review-bot state, even for a change this small. Flagged in the PR description with the same detail you found (`shape[0] == 1` + mismatched second dim → `RuntimeError` from `.expand()` instead of the documented `ValueError`) so it's not lost. The test here documents the current behavior rather than the ideal one, per your suggestion I dropped the "known mismatch" framing from the test comment and pointed it at the PR description instead.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (maxime-leiber) [reply tests/test_core.py:122]: No need for a separate issue — it's a minor, unreachable-in-practice edge case (no current caller passes an inconsistent shape here), and it's already captured in the PR description for Maxime to weigh whenever `_core.py` numeric-path changes are next on her list. Thanks for double-checking though.

---
_Generated by [Claude Code](https://claude.ai/code)_
